vms/platformvm/state/camino_dao_proposal.go

```diff
@@ -0,0 +1,142 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm/blocks"
+	"github.com/ava-labs/avalanchego/vms/platformvm/dao"
+)
+
+type ProposalLookup struct {
+	Proposal *dao.Proposal
```

Comment:
not sure if you got this already on your radar, but serialization tags are missing here for the ProposalLookup struct fields.